pkg/codec/example_test.go

```diff
@@ -0,0 +1,177 @@
+package codec_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"reflect"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/codec"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+// This example demonstrates how you can have config for itemTypes with one codec,
+// one that is modified before encoding for on-chain and modified after decoding for off-chain
+// the other is left unmodified during encoding and decoding.
+const (
+	anyUnmodifiedTypeName     = "Unmodified"
+	anyModifiedStructTypeName = "SecondItem"
+)
+
+var _ types.RemoteCodec = &ExampleStructJsonCodec{}
+
+type ExampleStructJsonCodec struct{}
+
+func (j ExampleStructJsonCodec) Encode(_ context.Context, item any, _ string) ([]byte, error) {
```

Comment:
Could drop the method unused method receivers?
```suggestion
func (ExampleStructJsonCodec) Encode(_ context.Context, item any, _ string) ([]byte, error) {
```